dashboard/src/components/RelativeDate/RelativeDate.js

```diff
@@ -67,10 +67,10 @@ class RelativeDate extends React.Component {
 
   render() {
     const { dateTime, capitalize, unit, style, ...props } = this.props;
-    const { now } = this.state;
 
     const formatter = new IntlRelativeFormat("en", { style });
     const dateValue = new Date(dateTime);
+    const now = new Date();
```

Comment:
This makes render impure. The correct way to have the component always render relative to the current date is to store `now` in state and update it on a timer.

It might be useful to create a util helper component to provide the current date to its children:

```jsx
<CurrentDateProvider interval={1000}>
 {(now) => <RelativeDate now={now} dateTime={dateTime}/>}
</CurrentDateProvider>
```

This could solve the time-shift problem for any component that references the current date.

Reply:
good idea 👍

Reply:
updated